plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -109,12 +108,12 @@ internal class LocationPuckManager(
   }
 
   fun updateCurrentPosition(vararg points: Point, options: (ValueAnimator.() -> Unit)? = null) {
-    val targets = if (lastLocation == null) arrayOf(*points, *points) else {
+    val targets = lastLocation?.let {
       show()
-      arrayOf(lastLocation, *points)
-    }
+      arrayOf(it, *points)
+    } ?: arrayOf(*points, *points)
```

Comment:
is this actually part of the fix as well?

Reply:
No, actually. Just removed unsafe cast that could possibly break if concurrency would take place